src/openzaak/components/zaken/tests/test_zaken.py

```diff
@@ -714,6 +715,88 @@ def test_create_zaak_with_incorrect_local_zaaktype(self):
+
+        data = {"bronorganisatie": "111222333"}
+
+        response = self.client.post(
+            reverse("zaak-reserveer_zaaknummer"), data, **ZAAK_WRITE_KWARGS
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        self.assertEqual(Zaak.objects.count(), 0)
+        self.assertEqual(ZaakIdentificatie.objects.count(), 1)
+
+        zaak_identificatie = ZaakIdentificatie.objects.get()
+        self.assertEqual(
+            response.json(), {"zaaknummer": zaak_identificatie.identificatie}
+        )
+
+        zaak_id = response.json()["zaaknummer"]
+        zaak_data = {
+            "zaaktype": f"http://testserver{self.zaaktype_url}",
+            "identificatie": zaak_id,
+            "bronorganisatie": "111222333",
+            "verantwoordelijkeOrganisatie": "111222333",
+            "startdatum": "2025-02-05",
+        }
+
+        create_response = self.client.post(
+            reverse("zaak-list"), zaak_data, **ZAAK_WRITE_KWARGS
+        )
+        self.assertEqual(create_response.status_code, status.HTTP_201_CREATED)
+
+        self.assertEqual(Zaak.objects.count(), 1)
+        self.assertEqual(ZaakIdentificatie.objects.count(), 1)
+
+        # try to reuse id
+        zaak_data_2 = {
+            "zaaktype": f"http://testserver{self.zaaktype_url}",
+            "identificatie": zaak_id,
+            "bronorganisatie": "111222333",
+            "verantwoordelijkeOrganisatie": "111222333",
+            "startdatum": "2025-02-07",
+        }
+
+        create_response_2 = self.client.post(
+            reverse("zaak-list"), zaak_data_2, **ZAAK_WRITE_KWARGS
+        )
+        self.assertEqual(create_response_2.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_reserve_zaak_identity_with_different_bronorganisatie(self):
+
+        data = {"bronorganisatie": "111222333"}
+
+        response = self.client.post(
+            reverse("zaak-reserveer_zaaknummer"), data, **ZAAK_WRITE_KWARGS
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        self.assertEqual(Zaak.objects.count(), 0)
+        self.assertEqual(ZaakIdentificatie.objects.count(), 1)
+
+        reserved_zaak_id = ZaakIdentificatie.objects.get()
+
+        zaak_id = response.data["zaaknummer"]
+        zaak_data = {
+            "zaaktype": f"http://testserver{self.zaaktype_url}",
+            "identificatie": zaak_id,
+            "bronorganisatie": "517439943",
+            "verantwoordelijkeOrganisatie": "517439943",
+            "startdatum": "2025-02-04",
+        }
+
+        create_response = self.client.post(
+            reverse("zaak-list"), zaak_data, **ZAAK_WRITE_KWARGS
+        )
+        self.assertEqual(create_response.status_code, status.HTTP_201_CREATED)
+
+        self.assertEqual(Zaak.objects.count(), 1)
+        self.assertEqual(ZaakIdentificatie.objects.count(), 2)
+
+        zaak = Zaak.objects.get()
+        self.assertNotEqual(zaak.identificatie, reserved_zaak_id)
```

Comment:
You're now comparing the identificatie string (e.g. `ZAAK-2025-0000000001`) to a `ZaakIdentificatie` object. I think this should probably be two assertions, because the zaaknummer is the same, but the `ZaakIdentificatie` object is different (because of the different organisatie):

```python
self.assertEqual(zaak.identificatie, zaak_id)
self.assertNotEqual(zaak.identificatie_ptr, reserved_zaak_id)
```